be/src/block_cache/starcache/block_file.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "block_cache/starcache/block_file.h"
+
+#include <glog/logging.h>
+
+#include "block_cache/starcache/aux_funcs.h"
+#include "block_cache/starcache/common/macros.h"
+
+namespace starrocks::starcache {
+
+Status BlockFile::open(bool pre_allocate) {
+    // TODO: use direct io (the buffer memory should be aligned)
+    // _fd = ::open(_file_path.c_str(), O_RDWR | O_CREAT | O_DIRECT, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+    int oflag = O_RDWR | O_CREAT;
+    if (!config::FLAGS_enable_os_page_cache) {
+        oflag |= O_DIRECT;
+    }
+    _fd = ::open(_file_path.c_str(), oflag, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+    if (_fd < 0) {
+        return _report_io_error("fail to open block file");
+    }
+
+    if (pre_allocate) {
+        if (::ftruncate(_fd, _quota_bytes) != 0) {
+            return _report_io_error("fail to truncate block file");
+        }
+    }
+    LOG(INFO) << "open block file success, fd: " << _fd << ", path: " << _file_path;
+    return Status::OK();
+}
+
+Status BlockFile::close() {
+    if (::close(_fd) != 0) {
+        return _report_io_error("fail to close block file");
+    }
+    return Status::OK();
+}
+
+[[maybe_unused]] static void deleter(void* buf) {
+    delete[] reinterpret_cast<char*>(buf);
+}
+
+Status BlockFile::write(off_t offset, const IOBuf& buf) {
+    ssize_t ret = 0;
+    size_t block_num = buf.backing_block_num();
+
+    void* aligned_data = nullptr;
+    size_t aligned_size = buf.size();
+    if (block_num == 1) {
+        auto data = buf.backing_block(0).data();
+        if (mem_need_align(data, buf.size())) {
+            aligned_size = align_buf(buf, &aligned_data);
+            ret = ::pwrite(_fd, aligned_data, aligned_size, offset);
+            free(aligned_data);
+        } else {
+            ret = ::pwrite(_fd, data, aligned_size, offset);
+        }
+    } else if (!config::FLAGS_enable_os_page_cache) {
```

Comment:
is that a new coding style? I have not seen this `FLAGS_` before.

Reply:
> is that a new coding style? I have not seen this `FLAGS_` before.

The original gflag variables have `FLAGS_` variable, while starrocks project does some convert work for the variable names. As we try to decouple the starcache module with starrocks project as much as posible, so that it can be splitted easily to a new registry in the future. So we use the standard gflags variable style and avoid depending on the starrocks project.